applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationClientTest.java

```diff
@@ -137,14 +142,17 @@ private void cleanUpResources() {
                     logger.info("Deleting key:label [{}:{}]. isLocked? {}", configurationSetting.key(), configurationSetting.label(), configurationSetting.isLocked());
 
                     if (configurationSetting.isLocked()) {
-                        return client.unlockKeyValue(configurationSetting.key(), configurationSetting.label(), null).flatMap(response -> {
+                        return client.unlockKeyValue(configurationSetting.key(), configurationSetting.label()).flatMap(response -> {
                             ConfigurationSetting kv = response.body();
-                            return client.deleteKeyValue(kv.key(), kv.label(), null);
+                            return client.deleteKeyValue(kv.key(), kv.label(), null)
+                                    .retryBackoff(3, Duration.ofSeconds(10));
```

Comment:
Is there a reason why retry after 10ms? If throttling is a concern, the service provides in the response `retry-after-ms` which will let you know how much time to wait for .
https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService?path=%2Fdocs%2Fprotocol%2Fthrottling.md&version=GBmaster

Reply:
Yeah. This is a feature that has yet to be implemented. :( #3183 is created to address that.